src/widgets/scrollbar.rs

```diff
@@ -124,6 +137,21 @@ pub enum ScrollbarOrientation {
 /// │  └──────── thumb
 /// └─────────── begin
 /// ```
+///
+/// # Examples
+///
+/// ```rust
+/// # use ratatui::prelude::*;
+/// # use ratatui::widgets::*;
+/// let items = vec![Line::from("Item 1"), Line::from("Item 2"), Line::from("Item 3")];
+/// let vertical_scroll = 0;
+/// let paragraph = Paragraph::new(items.clone()).scroll((vertical_scroll, 0));
+/// let scrollbar = Scrollbar::default()
+///     .orientation(ScrollbarOrientation::VerticalRight)
+///     .begin_symbol(Some("↑"))
+///     .end_symbol(Some("↓"));
+/// let scrollbar_state = ScrollbarState::new(items.iter().len() as u16).position(vertical_scroll);
+/// ```
```

Comment:
What about making this show the render as well? That's tripped up users previously (see #413).
(side note: perhaps the position etc. should be usize not u16?)

```suggestion
/// ```rust
/// # use ratatui::prelude::*;
/// # use ratatui::widgets::*;
/// # fn render_paragraph_with_scrollbar<B: Backend>(frame: &mut Frame<B>, area: Rect) {
/// let items = vec![
///     Line::from("Item 1"),
///     Line::from("Item 2"),
///     Line::from("Item 3"),
/// ];
/// let vertical_scroll = 0; // this would be passed from the app
/// let scrollbar = Scrollbar::default()
///     .orientation(ScrollbarOrientation::VerticalRight)
///     .begin_symbol(Some("↑"))
///     .end_symbol(Some("↓"));
/// let mut scrollbar_state = ScrollbarState::new(items.len() as u16).position(vertical_scroll);
///
/// let paragraph = Paragraph::new(items)
///     .block(Block::new().borders(Borders::RIGHT)) // to hold the scrollbar
///     .scroll((vertical_scroll, 0));
///
/// frame.render_widget(paragraph, area);
/// frame.render_stateful_widget(scrollbar, area, &mut scrollbar_state);
/// # }
/// ```
```

Reply:
I didn't realize you could put it in a function with a doc comment like that, that's neat. 

I actually considered it, but saw that all other widgets didn't have any rendering examples. I've added it in now though.

Reply:
> side note: perhaps the position etc. should be usize not u16?

I think so too. All `position`, `content_length` and `viewport_content_length` can be greater than the max u16 value. This would be a breaking change, would it not?

Reply:
yep